dali/imgcodec/parsers/webp_test.cc

```diff
@@ -39,33 +38,59 @@ class WebpParserTest : public ::testing::Test {
     EXPECT_FALSE(parser_.CanParse(&img));
   }
 
-  void TestValidImage(const std::string &filename, TensorShape<> expected_shape) {
-    auto img = ImageSource::FromFilename(webp_directory_ + filename);
+  void TestValidImage(const std::string &filename, TensorShape<> expected_shape,
+                      const Orientation expected_orientation = Orientation{}) {
+    auto img = ImageSource::FromFilename(filename);
     EXPECT_TRUE(parser_.CanParse(&img));
     EXPECT_EQ(parser_.GetInfo(&img).shape, expected_shape);
   }
+
+  void TestRotations(const std::string &filename_prefix, TensorShape<> expected_shape) {
+    TestValidImage(filename_prefix + "_original.webp",
+                   expected_shape, {0, false, false});
+    TestValidImage(filename_prefix + "_horizontal.webp",
+                   expected_shape, {0, false, false});
+    TestValidImage(filename_prefix + "_mirror_horizontal.webp",
+                   expected_shape, {0, true, false});
+    TestValidImage(filename_prefix + "_rotate_180.webp",
+                   expected_shape, {180, false, false});
+    TestValidImage(filename_prefix + "_mirror_vertical.webp",
+                   expected_shape, {0, false, true});
+    TestValidImage(filename_prefix + "_mirror_horizontal_rotate_270_cw.webp",
+                   expected_shape, {90, true, false});
+    TestValidImage(filename_prefix + "_rotate_90_cw.webp",
+                   expected_shape, {270, false, false});
+    TestValidImage(filename_prefix + "_mirror_horizontal_rotate_90_cw.webp",
+                   expected_shape, {270, true, false});
+    TestValidImage(filename_prefix + "_rotate_270_cw.webp",
+                   expected_shape, {90, false, false});
+  }
+
+  const std::string webp_directory_ = testing::dali_extra_path() + "/db/single/webp/";
+  const std::string orientation_directory_ = testing::dali_extra_path() +
+                                             "/db/imgcodec/webp/orientation/";
```

Comment:
Likewise, please use `make_string` for concatenation
```suggestion
  const std::string webp_directory_ = make_string(testing::dali_extra_path(), "/db/single/webp/");
  const std::string orientation_directory_ = make_string(testing::dali_extra_path(),
                                             "/db/imgcodec/webp/orientation/");
```

Reply:
Actually, some path joining function would be more useful - and a relative path shouldn't start or end with a `/`.

Reply:
Good point, `os.path.join` looks like a good candidate also :)